### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by jumaffre**
- Does not fail horribly if `metrics.yml` file does not exist.
- Create report graph on git environment (i.e. local env, not a PR) and compare to `master` branch by default, except if `CIMETRICS_TARGET_BRANCH` env var is set.
- Sort metrics on plot alphabetically (for now).

### CODE REVIEW

**Comment by ghost** [comment]
images

**Comment by ghost** [comment]
images

**Comment by ghost** [comment]
images

**Comment by ghost** [comment]
images

**Comment by ghost** [comment]
images

File: cimetrics/env.py [hunk 1]
```diff
@@ -85,8 +110,10 @@ def is_pr(self) -> bool:
 
     @property
     def target_branch(self) -> str:
-        assert self.is_pr
-        return os.environ["SYSTEM_PULLREQUEST_TARGETBRANCH"]
+        target_branch_ = os.environ.get("SYSTEM_PULLREQUEST_TARGETBRANCH")
```
Comment (achamayou) [inline cimetrics/env.py]: .get() takes a default, it’s shorter to pass it and return the result directly.

File: cimetrics/upload.py [hunk 2]
```diff
@@ -23,13 +23,24 @@ def publish(self):
...
         client = pymongo.MongoClient(self.env.mongo_connection)
-        db = client[self.env.mongo_db]
-        coll = db[self.env.mongo_collection]
+
+        db = None
+        coll = None
+        try:
+            db = client[self.env.mongo_db]
+            coll = db[self.env.mongo_collection]
+        except:
```
Comment (achamayou) [inline cimetrics/upload.py]: Never except:, except Exception at worst, but here we should really be more precise. We don’t want to hide a permission issue for example.

**Review by achamayou — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
images

**Comment by ghost** [comment]
images

**Comment by ghost** [comment]
images

**Comment by jumaffre** [comment]
Merging this now as I believe that the feedback has been addressed and some changes are necessary for CCF. Will continue further improvements/refactoring in further PRs.

**Pull request merged**